core/src/main/java/org/apache/calcite/rel/core/Aggregate.java

```diff
@@ -578,45 +582,42 @@ public static List<Integer> getRollup(List<ImmutableBitSet> groupSets) {
   public static class AggCallBinding extends SqlOperatorBinding {
     private final List<RelDataType> preOperands;
     private final List<RelDataType> operands;
-    private final int groupCount;
     private final boolean filter;
+    private final boolean hasEmptyGroup;
 
     /**
      * Creates an AggCallBinding.
      *
-     * @param typeFactory  Type factory
-     * @param aggFunction  Aggregate function
-     * @param preOperands  Data types of pre-operands
-     * @param operands     Data types of operands
-     * @param groupCount   Number of columns in the GROUP BY clause
-     * @param filter       Whether the aggregate function has a FILTER clause
+     * @param typeFactory   Type factory
+     * @param aggFunction   Aggregate function
+     * @param preOperands   Data types of pre-operands
+     * @param operands      Data types of operands
+     * @param hasEmptyGroup Whether the aggregate has a empty group
+     * @param filter        Whether the aggregate function has a FILTER clause
      */
     public AggCallBinding(RelDataTypeFactory typeFactory,
         SqlAggFunction aggFunction, List<RelDataType> preOperands,
-        List<RelDataType> operands, int groupCount,
```

Comment:
These constructors are all public and their input parameters have been modified. Consider creating an deprecated constructor with the same input parameters as the original construction method.